pkg/retry/azure_retry.go

```diff
@@ -162,18 +162,29 @@ func doBackoffRetry(s autorest.Sender, r *http.Request, backoff *Backoff) (resp
 			return
 		}
 		resp, err = s.Do(rr.Request())
-		rerr := GetErrorWithRetriableHTTPStatusCodes(resp, err, backoff.RetriableHTTPStatusCodes)
+		rerr := GetError(resp, err)
 		// Abort retries in the following scenarios:
 		// 1) request succeed
-		// 2) request is not retriable
-		// 3) request has been throttled
-		// 4) request contains non-retriable errors
-		// 5) request has completed all the retry steps
+		// 2) request has been throttled
+		// 3) request is not retriable or contains non-retriable errors
+		// 4) request has completed all the retry steps
 		if rerr == nil {
 			return resp, nil
 		}
 
-		if !rerr.Retriable || rerr.IsThrottled() || backoff.isNonRetriableError(rerr) || backoff.Steps == 1 {
+		if rerr.IsThrottled() {
+			return resp, rerr.RawError
+		}
+
+		if !rerr.Retriable {
+			if IsInHTTPStatusCodeSet(rerr, backoff.RetriableHTTPStatusCodes) || isInErrorsSet(rerr, backoff.NonRetriableErrors) {
```

Comment:
why should it become retriable and continue when `isInErrorsSet(rerr, backoff.NonRetriableErrors)` ?

Reply:
updated ut and func in new commit. Thanks!